plasma_framework/contracts/src/framework/registries/ExitGameRegistry.sol

```diff
@@ -6,6 +6,11 @@ contract ExitGameRegistry is Operated {
     mapping(uint256 => address) private _exitGames;
     mapping(address => uint256) private _exitGameToTxType;
 
+    event ExitGameRegistered(
+        uint256 txType,
```

Comment:
I'am afraid that hardcoding one tx type per exit game would prevent us from the future upgrade as talked with @paulperegud. 

We would rather make the interface of such registration future proof -> at some point, we would like to prepare such an exit game that could serve for all tx types.

Reply:
well....I am not really sure how the "future proofed" exit game should exactly look like. And I guess what I have in my mind might be different from yours.

What I have in my mind is that even we have an interface (well...I think it would be an abstract contract actually, as we probably need some logic), each implementation of exit game would still be 1:1 mapping. It is just the core logic for plasma protocol need would be pre-written in a template/abstract contract.

However, I am very curious what is the construction in you and pepesza's mind?